src/vs/workbench/parts/files/browser/views/explorerViewer.ts

```diff
@@ -557,6 +557,134 @@ export class FileSorter implements ISorter {
 	}
 }
 
+// Mixed Sorter
+export class MixedSorter implements ISorter {
+
+	public compare(tree: ITree, statA: FileStat, statB: FileStat): number {
+		if (statA instanceof NewStatPlaceholder) {
+			return -1;
+		}
+
+		if (statB instanceof NewStatPlaceholder) {
+			return 1;
+		}
+
+		// Do not sort roots
+		if (statA.isRoot) {
+			return -1;
+		}
+
+		if (statB.isRoot) {
+			return 1;
+		}
+
+		return comparers.compareFileNames(statA.name, statB.name);
+	}
+}
+
+// FilesFirst Sorter
+export class FilesFirstSorter implements ISorter {
+
+	public compare(tree: ITree, statA: FileStat, statB: FileStat): number {
+		if (statA.isDirectory && !statB.isDirectory) {
+			return 1;
+		}
+
+		if (statB.isDirectory && !statA.isDirectory) {
+			return -1;
+		}
+
+		if (statA instanceof NewStatPlaceholder) {
+			return -1;
+		}
+
+		if (statB instanceof NewStatPlaceholder) {
+			return 1;
+		}
+
+		// Do not sort roots
+		if (statA.isRoot) {
+			return -1;
+		}
+
+		if (statB.isRoot) {
+			return 1;
+		}
+
+		return comparers.compareFileNames(statA.name, statB.name);
+	}
+}
+
+// Type Sorter
+export class TypeSorter implements ISorter {
+
+	public compare(tree: ITree, statA: FileStat, statB: FileStat): number {
+		if (statA.isDirectory && !statB.isDirectory) {
+			return -1;
+		}
+
+		if (statB.isDirectory && !statA.isDirectory) {
+			return 1;
+		}
+
+		if (statA instanceof NewStatPlaceholder) {
+			return -1;
+		}
+
+		if (statB instanceof NewStatPlaceholder) {
+			return 1;
+		}
+
+		// Do not sort roots
+		if (statA.isRoot) {
+			return -1;
+		}
+
+		if (statB.isRoot) {
+			return 1;
+		}
+
+		return comparers.compareFileExtensions(statA.name, statB.name);
+	}
+}
+
+// Last modified date Sorter
+export class ModifiedSorter implements ISorter {
+
+	public compare(tree: ITree, statA: FileStat, statB: FileStat): number {
+		if (statA.isDirectory && !statB.isDirectory) {
+			return -1;
+		}
+
+		if (statB.isDirectory && !statA.isDirectory) {
+			return 1;
+		}
+
+		if (statA instanceof NewStatPlaceholder) {
+			return -1;
+		}
+
+		if (statB instanceof NewStatPlaceholder) {
+			return 1;
+		}
+
+		// Do not sort roots
+		if (statA.isRoot) {
+			return -1;
+		}
+
+		if (statB.isRoot) {
+			return 1;
+		}
+
+		if (statA.mtime === statB.mtime) {
+			return 0;
```

Comment:
Wouldn't it make sense here to fallback to the default sorting by file name in case the modified time is the same?